src/detect-filestore.c

```diff
@@ -264,6 +264,23 @@ static int DetectFilestoreMatch (ThreadVars *t, DetectEngineThreadCtx *det_ctx,
      * matches. */
     if (file != NULL) {
         file_id = file->file_store_id;
+
+        if (s->id > 0) {
+            if (file->sid_cnt >= file->sid_max) {
+                void* p = SCRealloc(file->sid, sizeof(uint32_t) * (file->sid_max + 8));
+                if (p == NULL){
+                    SCFree(file->sid);
+                    file->sid = NULL;
+                    file->sid_cnt = 0;
+                    file->sid_max = 0;
+                    SCReturnInt(0);
```

Comment:
while the chance of this failure occurring is very low, it still seems wrong to return 0 (no match) here. The error should be handled such that it won't affect the actual matching process.